torchrl/data/replay_buffers/replay_buffers.py

```diff
@@ -757,13 +774,16 @@ def update_tensordict_priority(self, data: TensorDictBase) -> None:
         if not isinstance(self._sampler, PrioritizedSampler):
             return
         if data.ndim:
-            priority = torch.tensor(
-                [self._get_priority(td) for td in data],
-                dtype=torch.float,
-                device=data.device,
-            )
+            if isinstance(data, LazyStackedTensorDict):
+                priority = torch.tensor(
+                    [self._get_priority_item(td) for td in data],
+                    dtype=torch.float,
+                    device=data.device,
+                )
```

Comment:
I guess we assume that the stack dim is 0 but it could not be (?)
I think we can consider the priority can be stacked, no? At the end of the day it's supposed to be one priority per item. Maybe I'm missing something

Reply:
this was the previous treatment so I am not super sure what was going on.

my guess is that it is assuming 0 as stack dim because in `expand` it stacks on 0.

are you suggesting to completely remove this and always go vectorized? I am down to try.

Reply:
maybe it was this way because the priority can have different shapes along the stack dim?

That is the only explanation I can guess